velox/expression/Expr.h

```diff
@@ -63,6 +63,62 @@ struct ExprStats {
   }
 };
 
+/// Data needed to generate exception context for the top-level expression.
+class ExprExceptionContext {
+ public:
+  ExprExceptionContext(
+      const Expr* FOLLY_NONNULL expr,
+      const RowVector* FOLLY_NONNULL vector)
+      : expr_(expr), vector_(vector) {}
+
+  void storeVector(
```

Comment:
Would it make sense to combine storeVector and storeSql methods into one method captureDataAndSql?

Why do we pass vector and expr to these methods? Aren't these available as member variables?